cmd/eskip/readclient.go

```diff
@@ -0,0 +1,127 @@
+package main
+
+import (
+	"github.com/zalando/skipper/eskip"
+	"github.com/zalando/skipper/eskipfile"
+	etcdclient "github.com/zalando/skipper/etcd"
+	innkeeperclient "github.com/zalando/skipper/innkeeper"
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+type ReadClient interface {
+	LoadAndParseAll() ([]*eskip.RouteInfo, error)
+
+	LoadUpdate() (eskip.RouteList, []string, error)
+}
+
+type stdinReader struct {
+	reader io.Reader
+}
+
+type inlineReader struct {
+	routes string
+}
+
+type idsReader struct {
+	ids []string
+}
+
+// TODO eliminate duplicate code in write and read clients
+func createReadClient(m *medium) (ReadClient, error) {
```

Comment:
I am really not a fan of applying this type of polymorphism in this case. Meaning the ReadClient and its implementations. The reason is that this doesn't make the code simpler but rather the opposite. The reasons why I don't see any benefit of it:
- these are stateless objects for a simple logic, within a single function that is called and the object is thrown away
- the switch may be hidden here, but it is still here
- with all the 'implementing' types, and their methods, the code just gets much more verbose